sensing/pointcloud_preprocessor/src/crop_box_filter/crop_box_filter_nodelet.cpp

```diff
@@ -143,7 +143,12 @@ void CropBoxFilterComponent::faster_filter(
                            point[1] > param_.min_y && point[1] < param_.max_y &&
                            point[0] > param_.min_x && point[0] < param_.max_x;
     if ((!param_.negative && point_is_inside) || (param_.negative && !point_is_inside)) {
-      memcpy(&output.data[output_size], &point, input->point_step);
+      memcpy(&output.data[output_size], &input->data[global_offset], input->point_step);
```

Comment:
The old `memcpy` was certainly wrong, because `point` is a `Eigen::Vector4f`, whose memory layout is not necessarily the same than `output.data`. However with the fixed `memcpy` why do we need to copy points xyz manually after? The `memcpy` line should be enough no?

Reply:
This is because `input` contains the data before the coordinate transformation, so the data after the transformation must be overwritten later.

Reply:
In that case would it be faster if we add the copy in a `if (transform_info.need_transform) {}` ?
the "fast" version would be something like:
```c++
if (transform_info.need_transform) {
  for (...) {
    Eigen::Vector4f point(...);
    // transform then memcpy and transformed xyz copy if inside
  }
} else {
  for (...) {
    Eigen::Vector4f point(...);
    // memcpy only if inside
  }
}
```